File: src/medium/t2/solve.go
package main

import "fmt"

func main() {
	prices := []int{7, 1, 5, 3, 6, 4}
	profit := maxProfit(prices)
	fmt.Println(profit)
}

func maxProfit(prices []int) int {
	total := 0
	for i := 1; i < len(prices); i++ {
		if prices[i] > prices[i-1] {
			total += prices[i] - prices[i-1]
		}
	}
	return total
}

File: src/medium/t2/solve.rs
pub fn max_profit(prices: Vec<i32>) -> i32 {
    let mut total = 0;
    for i in 1..prices.len() {
        if prices[i] > prices[i - 1] {
            total += prices[i] - prices[i - 1]
        }
    }
    total
}

#[cfg(test)]
mod test {
    use super::max_profit;

    #[test]
    fn easy_t1_solve() {}
}
